File: SortingTechniques/SelectionSort/SelectionSort.java
package SortingTechniques.SelectionSort;

import java.util.Arrays;

public class SelectionSort {
    public static void main(String args[]){
        int[] arr = {3, 2, 5, 7, 1, 8, 9};
        selectionSort(arr);
        System.out.println(Arrays.toString(arr));
    }

    public static void selectionSort(int arr[]){
        for(int i = 0; i < arr.length; i++){
            int last = arr.length - 1 - i;
            int maxIndex = getMaxIndex(arr, 0, last);

            swap(arr, maxIndex, last);
        }
    }

    static void swap(int arr[], int first, int second){
        int temp = arr[first];
        arr[first] = arr[second];
        arr[second] = temp;
    }

    static int getMaxIndex(int arr[], int start, int end){
        int max = start;
        for(int i = start; i <= end; i++){
            if(arr[max] < arr[i]){
                max = i;
            }
        }
        return max;
    }
}

File: SortingTechniques/SelectionSort.c
#include <stdio.h>
#include <stdlib.h>

void swap(int *a , int *b){
    int temp = *a;
    *a = *b; *b = temp;
}

void selectionSort(int a[], int n){
    int i,j,k;
    for( i = 0; i < n-1; i++){
        j=i;
        k=i;
        for(j; j < n; i++){
            if(a[j]< a[k]){
                k=j;
            }
        }
        swap(&a[i],&a[k]);
    }

}

int main(){
    int n;
    scanf("%d",&n);
    int arr[n];
    for(int i = 0; i < n; i++){
        scanf("%d", &arr[i]);
    }

    selectionSort(arr, n);
    for(int i = 0; i < n; i++){
        printf("%d,",arr[i]);
    }

}
